nix/marlowe-symbolic-lambda.nix

```diff
@@ -0,0 +1,41 @@
+{ pkgs, haskellPackages }:
```

Comment:
I'd probably put this in `marlowe-symbolic`. I think it's good to have nix files for building stuff related to a project in that subdirectory, and the `nix` folder is mostly for stuff that has no other home (e.g. building out-of-tree stuff).